### DISCUSSION TITLE
the staticfile finders cannot find my component js and css files

### DISCUSSION THREAD

**Discussion by simkimsia** [django-components/django-components#160] [Q&A] [answered]
This is the errors 

<img width="872" alt="image" src="https://user-images.githubusercontent.com/245021/179350193-5042431c-e1b2-4bb5-9c75-950b0cee25f3.png">

This is my folder structure.

My app name is `hypercomponents`

<img width="443" alt="image" src="https://user-images.githubusercontent.com/245021/179350203-c35439cf-cf0b-4da1-9039-5a30596e31cd.png">

This is my components.py

```python
@component.register("calendar")
class Calendar(component.Component):
    """
    Note that Django will look for templates inside `[your app]/components` dir
    To customize which template to use based on context override get_template_name instead
    """
    template_name = "django_components/00_calendar/calendar.html"


    def get_context_data(self, date):
        """
        This component takes one parameter, a date string to show in the template
        """
        return {
            "date": date,
        }

    class Media:
        """
        Media class is used to define the css and js files that will be used in the template
        """
        css = 'hypercomponents/components/django_components/00_calendar/calendar.css'
        js = 'hypercomponents/components/django_components/00_calendar/calendar.js'
```

the component is loading the markup properly

This is my static settings

```python
# STATIC
# ------------------------------------------------------------------------------
# https://docs.djangoproject.com/en/dev/ref/settings/#static-root
STATIC_ROOT = str(ROOT_DIR / "staticfiles")
# https://docs.djangoproject.com/en/dev/ref/settings/#static-url
STATIC_URL = "/static/"
# https://docs.djangoproject.com/en/dev/ref/contrib/staticfiles/#std:setting-STATICFILES_DIRS
STATICFILES_DIRS = [str(APPS_DIR / "static")]
# https://docs.djangoproject.com/en/dev/ref/contrib/staticfiles/#staticfiles-finders
STATICFILES_FINDERS = [
    "django.contrib.staticfiles.finders.FileSystemFinder",
    "django.contrib.staticfiles.finders.AppDirectoriesFinder",
]
```

Please advise.

**Comment by EmilStenstrom** [comment]
As you see from the paths that gives 404, it tries to find the media in the **static** directory. 

Two things to try (this is really django-specific, and has nothing to do with django_components):

1. Try moving your static files to the global static directory in your project
2. Try moving your files inside a static directory.

**Reply by simkimsia** [reply]
How would you go about organizing the static files of the individual components in a real Django project?

**Comment by EmilStenstrom** [comment]
@simkimsia I've now added several things that will simplify setup:

1. A project called "sampleproject" that shows how to setup django_components.
2. Improved instructions in the README with a screenshot of the recommended directory structure
3. Added support for automatically detecting more component directory locations, and autoloading from there
4. Added a part about STATICFILES_DIRS in the setup instructions (this is likely your problem).

Please update to the latest version of the library, and go through the instructions again and see if you can get things working.

**Reply by simkimsia** [reply]
Thank you i will give this a try and respond later this week.

**Reply by simkimsia** [reply]
@EmilStenstrom 

Can I ask how should I "install" the sample project?

I install components as a pypi into a sandbox django project to play

Do I git clone components and treat it as a separate django project?


Or git clone but then copy out the sampleproject and put it into my sandbox and use it as django app?

**Reply by EmilStenstrom** [reply]
You can decide. I think the easiest is to clone this repo, cd into the sampleproject repo, make sure django and django_components are installed, and start the dev server.

**Reply by simkimsia** [reply]
Thank you @EmilStenstrom  overall, i figured out the steps to make this work. I have also added a PR #164 that i discover as a response to the issue in #163 while trying to make this work.

Thank you for being so responsive.

**Comment by simkimsia** [comment, marked as answer]
In terms of specifically and narrowly answer this question, I think the solution is

1. upgrade django-components to at least 0.21 and above.
2. configure the STATICFILES_DIR according to how the components are structured in your own project

then the solution is fixed.
